cashu/wallet/helpers.py

```diff
@@ -0,0 +1,453 @@
+            )
+        else:
+            # no mint information present, we extract the proofs and use wallet's default mint
+
+            proofs = [p for t in tokenObj.token for p in t.proofs]
+            # first we load the mint URL from the DB
+            keyset_in_token = proofs[0].id
+            assert keyset_in_token
+            # we get the keyset from the db
+            mint_keysets = await get_keyset(id=keyset_in_token, db=wallet.db)
+            if is_api:
+                assert mint_keysets, HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="we don't know this keyset",
+                )
+                assert mint_keysets.mint_url, HTTPException(
+                    status_code=status.HTTP_400_BAD_REQUEST,
+                    detail="we don't know this mint's URL",
+                )
+            else:
+                assert mint_keysets, Exception("we don't know this keyset")
+                assert mint_keysets.mint_url, Exception("we don't know this mint's URL")
+            # now we have the URL
+            mint_wallet = Wallet(
+                mint_keysets.mint_url,
+                os.path.join(settings.cashu_dir, wallet.name),
+            )
+            await mint_wallet.load_mint(keyset_in_token)
+            _, _ = await mint_wallet.redeem(proofs, script, signature)
+            if not is_api:
+                print(f"Received {sum_proofs(proofs)} sats")
+
+    # reload main wallet so the balance updates
+    await wallet.load_proofs()
+    if is_api:
+        return wallet.available_balance
+    else:
+        wallet.status()
+
+
+async def send(
+    wallet: Wallet, amount: int, lock: str, legacy: bool, is_api: bool = False
+):
+    """
+    Prints token to send to stdout.
+    """
+    if lock and len(lock) < 22:
+        if is_api:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail="lock has to be at least 22 characters long.",
+            )
+        else:
+            print("Error: lock has to be at least 22 characters long.")
+            return
+    p2sh = False
+    if lock and len(lock.split("P2SH:")) == 2:
+        p2sh = True
+
+    wallet = await get_mint_wallet(wallet, is_api=is_api)
+    await wallet.load_proofs()
+
+    _, send_proofs = await wallet.split_to_send(
+        wallet.proofs, amount, lock, set_reserved=True
+    )
+    token = await wallet.serialize_proofs(
+        send_proofs,
+        include_mints=True,
+    )
+    if not is_api:
+        print(token)
+
+    if legacy:
+        if not is_api:
+            print("")
+            print("Old token format:")
+            print("")
+        token = await wallet.serialize_proofs(
+            send_proofs,
+            legacy=True,
+        )
+        if not is_api:
+            print(token)
+
+    if is_api:
+        return wallet.available_balance, token
```

Comment:
All prints can remain also for `is_api`.

Ideally, there should be no `is_api` in here since it hopefully doesn't change the business logic anyway.